core/src/main/java/org/apache/iceberg/ManifestLists.java

```diff
@@ -21,51 +21,79 @@
 import java.io.IOException;
 import java.util.List;
 import org.apache.iceberg.avro.Avro;
+import org.apache.iceberg.avro.AvroIterable;
 import org.apache.iceberg.exceptions.RuntimeIOException;
 import org.apache.iceberg.io.CloseableIterable;
 import org.apache.iceberg.io.InputFile;
 import org.apache.iceberg.io.OutputFile;
+import org.apache.iceberg.relocated.com.google.common.annotations.VisibleForTesting;
 import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
 import org.apache.iceberg.relocated.com.google.common.collect.Lists;
 
 class ManifestLists {
   private ManifestLists() {}
 
   static List<ManifestFile> read(InputFile manifestList) {
-    try (CloseableIterable<ManifestFile> files =
-        Avro.read(manifestList)
-            .rename("manifest_file", GenericManifestFile.class.getName())
-            .rename("partitions", GenericPartitionFieldSummary.class.getName())
-            .rename("r508", GenericPartitionFieldSummary.class.getName())
-            .classLoader(GenericManifestFile.class.getClassLoader())
-            .project(ManifestFile.schema())
-            .reuseContainers(false)
-            .build()) {
-
+    try (CloseableIterable<ManifestFile> files = manifestFileIterable(manifestList)) {
       return Lists.newLinkedList(files);
-
     } catch (IOException e) {
       throw new RuntimeIOException(
           e, "Cannot read manifest list file: %s", manifestList.location());
     }
   }
 
+  @VisibleForTesting
+  static AvroIterable<ManifestFile> manifestFileIterable(InputFile manifestList) {
+    return Avro.read(manifestList)
+        .rename("manifest_file", GenericManifestFile.class.getName())
+        .rename("partitions", GenericPartitionFieldSummary.class.getName())
+        .rename("r508", GenericPartitionFieldSummary.class.getName())
+        .classLoader(GenericManifestFile.class.getClassLoader())
+        .project(ManifestFile.schema())
+        .reuseContainers(false)
+        .build();
+  }
+
   static ManifestListWriter write(
       int formatVersion,
       OutputFile manifestListFile,
       long snapshotId,
       Long parentSnapshotId,
       long sequenceNumber) {
+    return write(
+        formatVersion,
+        manifestListFile,
+        snapshotId,
+        parentSnapshotId,
+        sequenceNumber,
+        /* compressionCodec */ null,
+        /* compressionLevel */ null);
```

Comment:
Style: Comments like this usually follow the argument.

Reply:
Removed the comments, as they aren't strictly necessary.